sky/packages/sky/lib/painting/paragraph_painter.dart

```diff
@@ -0,0 +1,147 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:sky' as sky;
+
+import 'package:sky/painting/text_style.dart';
+
+// This must be immutable, because we won't notice when it changes
+abstract class TextSpan {
+  sky.Node _toDOM(sky.Document owner);
+  String toString([String prefix = '']);
+
+  void _applyStyleToContainer(sky.Element container) {
```

Comment:
Yes. (Actually this code existed before this pull request, I just moved it around.)